src/slack_block_util.ts

```diff
@@ -412,45 +597,66 @@ export function replaceVoterPanelBlocks(
   return newBlocks;
 }
 
-// This function mutates the blocks input.
-export function populateDropdownNewInitialValue(
+// This function finds the element for a given action ID in a set of blocks
+export function findElementWithActionId(
   blocks: SlackBlock[],
-  actionId: SlackActionId,
-  newInitialValue: VoterStatus
-): boolean {
+  actionId: string
+): SlackElement | null {
   for (const i in blocks) {
     const block = blocks[i];
     if (block.type === 'actions') {
       const elements = block.elements;
       for (const j in elements) {
         const element = elements[j];
         if (element.action_id === actionId) {
-          if (element.type === 'static_select') {
-            // Assume new options is already in the list of old options
-            const newOption = element.options.find(
-              (o: SlackOption) => o.value === newInitialValue
-            );
-            if (!newOption) {
-              logger.error(
-                `Option with value ${newInitialValue} was not found`
-              );
-              return false;
-            }
-
-            element.initial_option = newOption;
-            // Javascript modifies the blocks by reference, return success
-            return true;
-          }
-          if (element.type === 'users_select') {
-            element.initial_user = newInitialValue;
-            // Javascript modifies the blocks by reference, return success
-            return true;
-          }
```

Comment:
This stuff just got moved below. It separates out finding an element by action ID from mutating it.